src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax;
+
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(Map<String, Integer> storage, FruitTransaction transaction) {
+        storage.merge(
+                transaction.getFruit(),
+                -transaction.getQuantity(),
+                Integer::sum
+        );
+    }
+}
```

Comment:
According to the checklist, handling the Purchase operation requires checking the result balance before saving it in the Storage - it should be positive. Throw a `RuntimeException` in case the balance is negative. Currently, your implementation in `apply` method does not handle the case when the result of a purchase operation would lead to a negative balance in the storage.